### ISSUE TITLE
Add 'Snap' to FreeCAD 'About' info

### ISSUE THREAD

**Issue by luzpaz** [FreeCAD/FreeCAD-snap#16]
Something that I've noticed that I was wondering is possible...?

In the [About](https://wik.freecad.org/About) info, it would be nice to indicate that we're using a Snap package + bonus name of the channel we're using (stable, beta, edge, misc-experimental). Is that something we can do?

**Comment by ppd** [comment]
The info for the about dialog is built here: https://github.com/FreeCAD/FreeCAD/blob/master/src/Gui/Splashscreen.cpp
If our extra data fits in there somehow, we can inject it into the config. If not, we'd need to extend the  about dialog upstream to support packaging information.

**Comment by luzpaz** [comment]
@chennes what do you think?

**Comment by chennes** [comment]
Sure, if something that indicates this is a Snap can be injected into the config file, it's easy to add it to the About dialog.

**Comment by ppd** [comment]
We should be able to simply ship a branding.xml, no? Like e.g. realthunder does: https://github.com/realthunder/FreeCADMakeImage/blob/master/conda/branding/asm3/branding/branding.xml

**Comment by chennes** [comment]
What environment variable indicates a Snap?

**Comment by ppd** [comment]
SNAP or SNAP_NAME would be good candidates: https://snapcraft.io/docs/environment-variables

**Comment by chennes** [comment]
Easy enough -- do any of those other variables provide any useful information for debugging? SNAP_REVISION? Or is it enough to know that this is a Snap?

**Comment by ppd** [comment]
SNAP_REVISION would be fantastic because it is unique and tells us the exact build the user has.

**Comment by chennes** [comment]
OK: I can't test it, but this should do the trick... https://github.com/FreeCAD/FreeCAD/commit/307cb3cda8c57c94336069fefd4c057f7aad91c1

**Comment by luzpaz** [comment]
Just triggered an 'edge' build. Will test and report back. Thanks chennes!

**Comment by luzpaz** [comment]
Didn't work AFAICT  
```
OS: Ubuntu Core 20 (KDE/plasma)
Word size of FreeCAD: 64-bit
Version: 0.20.28966 (Git) Snap 46
Build type: Release
Branch: master
Hash: 307cb3cda8c57c94336069fefd4c057f7aad91c1
Python 3.8.10, Qt 5.15.3, Coin 4.0.0, OCC 7.6.2
Locale: English/United States (en_US)
```

**Comment by luzpaz** [comment]
I was thinking that, IMHO, it is more appropriate to add that string in on the first line for `OS:`  
for example: `OS: Ubuntu Core 20 (KDE/plasma) Snap 46`

**Comment by ppd** [comment]
I'm not so sure. Imho, the snap revision is the build version. Therefore, it is logically related to the application version. See also how snap info freecad puts it right next to the application version.

Ubuntu Core 20 is the OS inside the container. I don't think that fits. And even worse: The core20 snap itself has a revision. A user might think that they're running core20 revision 46, not freecad revision 46.

tl/dr: Revision should not appear next to the OS version.
